### PR TITLE
Report on stale branches

### PR DESCRIPTION
**Pull request by sldblog**
## What does this pull request do?

Add a report that exposes how many unmerged branches exist (with commits that are not reachable from origin/{main,master}) and what is the oldest unmerged commit

## What is the intent behind these changes?

Expose how long code that might be useful on production sits around

This is **not** meant to generate blame, more just visibility of work that has been started or done but not yet in production

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/hmpps/architecture/documentation/codeTracker.kt [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    "branch",
+    "--remote",
+    "--list",
+    "--no-merged=origin/$mainBranch",
+    "--format=%(refname)",
+    "--sort=committerdate"
+  )
+    .directory(repoDir)
+    .start()
+    .also { it.waitFor(1, TimeUnit.SECONDS) }
```
Comment (mxco86) [inline src/main/kotlin/uk/gov/justice/hmpps/architecture/documentation/codeTracker.kt:69]: Why do we need this delay, out of interest?

File: src/main/kotlin/uk/gov/justice/hmpps/architecture/documentation/codeTracker.kt [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+  if (p.exitValue() != 0) {
+    val message = p.errorStream.bufferedReader().readText()
+    return if (message.contains("malformed object name origin/main")) {
+      remoteBranches(repoDir, "master")
```
Comment (mxco86) [inline src/main/kotlin/uk/gov/justice/hmpps/architecture/documentation/codeTracker.kt:74]: Quite annoying to have to do this but I'm not sure the alternatives are any better

**Review by mxco86 — APPROVED** [review]
Couple of queries but 👍

File: src/main/kotlin/uk/gov/justice/hmpps/architecture/documentation/codeTracker.kt (hunk 1 above)
Comment (sldblog) [reply src/main/kotlin/uk/gov/justice/hmpps/architecture/documentation/codeTracker.kt:69]: It's acting like a timeout, so we don't wait forever

File: src/main/kotlin/uk/gov/justice/hmpps/architecture/documentation/codeTracker.kt (hunk 2 above)
Comment (sldblog) [reply src/main/kotlin/uk/gov/justice/hmpps/architecture/documentation/codeTracker.kt:74]: I suppose I could use `HEAD` in the assumption that the default branch is representative of the latest deployment. Not sure if that's a fair assumption or not

**Comment by sldblog** [comment]
Thanks @mxco86!

**Pull request merged**